bootspec/src/v1.rs

```diff
@@ -37,9 +37,12 @@ pub struct GenerationV1 {
     /// Path to "append-initrd-secrets" script -- $toplevel/append-initrd-secrets
     pub initrd_secrets: Option<PathBuf>,
     /// Mapping of specialisation names to their boot.json
-    pub specialisation: HashMap<SpecialisationName, GenerationV1>,
+    pub specialisation: HashMap<SpecialisationName, GenerationV1<Extension>>,
     /// config.system.build.toplevel path
     pub toplevel: SystemConfigurationRoot,
+    /// User extensions for this specification
+    #[serde(default)]
+    pub extensions: Extension,
```

Comment:
Is there a reason why this needs to be a generic argument instead of just `pub extensions: HashMap<String, serde_json::Value>`?

Reply:
Oh, I see -- so downstream consumers with extensions can more easily validate that they match their expected structure, rather than having to re-parse or something.